android/src/main/java/com/reactnativecommunity/asyncstorage/AsyncStorageModule.java

```diff
@@ -44,6 +50,31 @@
   private ReactDatabaseSupplier mReactDatabaseSupplier;
   private boolean mShuttingDown = false;
 
+
+  private static final int CPU_COUNT = Runtime.getRuntime().availableProcessors();
+  private static final int CORE_POOL_SIZE = Math.max(2, Math.min(CPU_COUNT - 1, 4));
+  private static final int MAXIMUM_POOL_SIZE = CPU_COUNT * 2 + 1;
+  private static final int KEEP_ALIVE_SECONDS = 30;
+  public static final Executor THREAD_POOL_EXECUTOR;
+  private static final ThreadFactory sThreadFactory = new ThreadFactory() {
+    private final AtomicInteger mCount = new AtomicInteger(1);
+
+    public Thread newThread(Runnable r) {
+      return new Thread(r, "AsyncStorageTask #" + mCount.getAndIncrement());
+    }
+  };
+
+  private static final BlockingQueue<Runnable> sPoolWorkQueue =
+          new LinkedBlockingQueue<Runnable>(128);
+
+  static {
+    ThreadPoolExecutor threadPoolExecutor = new ThreadPoolExecutor(
```

Comment:
Should we be using one of the `Executors` factory methods as mentioned in [the documentation](https://developer.android.com/reference/java/util/concurrent/ThreadPoolExecutor) instead? The  [`Executors#newSingleThreadExecutor`](https://developer.android.com/reference/java/util/concurrent/Executors.html#newSingleThreadExecutor()) sounds like a good fit to me as it will effectively serialize all storage operations.

Reply:
@Krizzu, I think using one of the factory methods will vastly simplify this PR.

Reply:
Yup, I agree - sequential operations make more sense to be used here. 

@gaodeng Can you amend your changes?

Reply:
sure,thanks for the tip

Reply:
@gaodeng 谢谢大兄弟，终于解决了这个问题